LibreHardwareMonitorLib/Hardware/Ring0.cs

```diff
@@ -232,6 +175,74 @@ private static bool ExtractDriver(string filePath)
             return false;
         }
 
+        private static bool TryCreateOrOpenExistingMutex(string name, out Mutex mutex)
+        {
+#if NETFRAMEWORK
+            MutexSecurity mutexSecurity = CreateMutexSecurity(MutexRights.Synchronize | MutexRights.Modify);
+            return TryCreateOrOpenExistingMutex(name, mutexSecurity, out mutex);
+#else
+            mutex = null;
+            try
+            {
+                mutex = new Mutex(false, name);
+                return false;
+            }
+            catch (UnauthorizedAccessException)
+            {
+                try
+                {
+                    mutex = Mutex.OpenExisting(name);
+                    return true;
+                }
+                catch { }
+            }
+            return false;
+#endif
+        }
+
+#if NETFRAMEWORK
+        private static MutexSecurity CreateMutexSecurity(MutexRights eventRights)
+        {
+            MutexSecurity mutexSecurity = new MutexSecurity();
+            SecurityIdentifier identity = new SecurityIdentifier(WellKnownSidType.WorldSid, null);
+            AccessControlType type = AccessControlType.Allow;
+            mutexSecurity.AddAccessRule(new MutexAccessRule(identity, eventRights, type));
+
+            return mutexSecurity;
+        }
+
+        private static bool TryCreateOrOpenExistingMutex(string name, MutexSecurity mutexSecurity, out Mutex mutex)
+        {
+            mutex = null;
+
+
+            try
+            {
+                // if the CreateMutex call fails, the framework will attempt to use OpenMutex
+                // to open the named mutex requesting SYNCHRONIZE and MUTEX_MODIFY rights.
+                mutex = new Mutex(false, name, out _, mutexSecurity);
+                return true;
+            }
+            catch (WaitHandleCannotBeOpenedException)
+            {
+                // the mutex cannot be opened, probably because a Win32 object of a different
+                // type with the same name already exists.
+                return false;
+            }
+            catch (UnauthorizedAccessException)
+            {
+                // the mutex exists, but the current process or thread token does not
+                // have permission to open the mutex with SYNCHRONIZE | MUTEX_MODIFY rights.
+                return false;
```

Comment:
This entire block seems pointless as it's catched below anyway. You can rewrite this into comments per exception.

Reply:
It was there to separate different scenarios to handle something like dumping it out to a log/debug file. Since as it is now, it just hides the exception (which I consider is a design decision that would make troubleshooting issues hard).

Reply:
I removed the multiple catch statements, and consolidated it into a single one, with these as comments for any future needs.